File: SDKs/amazonaws.com/ds/2015-04-16/go/pkg/models/operations/getdirectorylimits.go
package operations

import (
"openapi/pkg/models/shared")


type GetDirectoryLimitsXAmzTargetEnum string

const (
    GetDirectoryLimitsXAmzTargetEnumDirectoryService20150416GetDirectoryLimits GetDirectoryLimitsXAmzTargetEnum = "DirectoryService_20150416.GetDirectoryLimits"
)


type GetDirectoryLimitsHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    XAmzTarget GetDirectoryLimitsXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
    
}

type GetDirectoryLimitsRequest struct {
    Headers GetDirectoryLimitsHeaders 
    Request map[string]interface{} `request:"mediaType=application/json"`
    
}

type GetDirectoryLimitsResponse struct {
    ClientException *interface{} 
    ContentType string 
    EntityDoesNotExistException *interface{} 
    GetDirectoryLimitsResult *shared.GetDirectoryLimitsResult 
    ServiceException *interface{} 
    StatusCode int64 
    
}

File: SDKs/amazonaws.com/ds/2015-04-16/typescript/src/sdk/models/operations/getdirectorylimits.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum GetDirectoryLimitsXAmzTargetEnum {
    DirectoryService20150416GetDirectoryLimits = "DirectoryService_20150416.GetDirectoryLimits"
}


export class GetDirectoryLimitsHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: GetDirectoryLimitsXAmzTargetEnum;
}


export class GetDirectoryLimitsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: GetDirectoryLimitsHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: Map<string, any>;
}


export class GetDirectoryLimitsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientException?: any;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  entityDoesNotExistException?: any;

  @SpeakeasyMetadata()
  getDirectoryLimitsResult?: shared.GetDirectoryLimitsResult;

  @SpeakeasyMetadata()
  serviceException?: any;

  @SpeakeasyMetadata()
  statusCode: number;
}
